docs/how-to-use-clusterctl.md

```diff
@@ -0,0 +1,81 @@
+
+`clusterctl` also comes with additional features. Example, `clusterctl` can also take in an optional
+`bootstrap-only-components.yaml` to provide resources to the bootstrap cluster without also providing them
+to the target cluster post-pivot.
+For more details about all the supported options run:
+
+```
+clusterctl create cluster --help
+```
+
+After generating the YAML files run the following command:
+
+```
+clusterctl create cluster --provider <PROVIDER> --bootstrap-type <BOOTSTRAP CLUSTER TYPE> -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+```
+
+Example usage:
+
+```
+# VMWare VSphere
+clusterctl create cluster --provider vsphere --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+
+# Amazon AWS
+clusterctl create cluster --provider aws --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+```
+
+**What happens when we run the command?**  
+After running the command first it creates a local cluster. If `kind` was passed as the `--bootstrap-type`
+it creates a local [kind](https://kind.sigs.k8s.io/) cluster. This cluster is generally referred to as the *bootstrap cluster*.  
+On this kind Kubernetes cluster the `provider-components.yaml` file is applied. This step loads the CRDs into
+the cluster. It also creates 2 [StatefulSet](https://Kubernetes.io/docs/concepts/workloads/controllers/statefulset/)
+pods that run the cluster api controller and the provider specific controller. These pods register the custom
+controllers that manage the newly defined resources (`Cluster`, `Machine`, `MachineSet`, etc).  
+
+Next, `clusterctl` applies the `cluster.yaml` and `machines.yaml` to the local kind Kubernetes cluster. This
+step creates a Kubernetes cluster with only a control-plane(as defined in `machines.yaml`) on the specified
+provider. This newly created cluster is generally referred to as the *management cluster* or *pivot cluster*
+or *initial target cluster*. The management cluster is responsible for creating and maintaining the work-load cluster.  
+  
+Lastly, `clusterctl` moves all the CRDs and the custom controllers from the bootstrap cluster to the
+management cluster and deletes the locally created bootstrap cluster. This step is referred to as the *pivot*.
+
+### Creating a workload cluster using the management cluster
+
+The *workload cluster* also sometimes referred to as the *target cluster* is the Kubernetes cluster on to which
+the final application is deployed. The target cluster is responsible for handling the workload of the application,
+not the management cluster.
+
+As the management cluster is up we can create a workload cluster by simply applying the appropriate
+`cluster.yaml`, `machines.yaml` and `machineset.yaml` on the management cluster. This will create the VMs(Nodes)
```

Comment:
You also need to apply addons for a CNI plugin; otherwise, your nodes will be `NotReady`.